### PR TITLE
Rename models 

### PR DESCRIPTION
**Pull request by sima-zhu**
Issues:
https://github.com/azure/azure-sdk-for-java/issues/11537
https://github.com/azure/azure-sdk-for-java/issues/11266

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexerAsyncClient.java [hunk 1]
```diff
@@ -233,19 +236,20 @@ public String getEndpoint() {
      * help with debugging.
      * @return a list of DataSources
      */
-    public PagedFlux<SearchIndexerDataSource> listDataSources(RequestOptions requestOptions) {
+    public PagedFlux<SearchIndexerDataSourceConnection> listDataSourceConnection(RequestOptions requestOptions) {
```
Comment (heaths) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexerAsyncClient.java:239]: This should be plural, i.e. `listDataSourceConnections`.

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexerAsyncClient.java:239]: Will check all naming and address in https://github.com/Azure/azure-sdk-for-java/pull/11590

**Pull request merged**